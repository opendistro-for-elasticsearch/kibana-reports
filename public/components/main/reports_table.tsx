/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React, { useState } from 'react';
import {
  EuiButton,
  // @ts-ignore
  EuiLink,
  EuiText,
  EuiIcon,
  EuiButtonIcon,
  EuiEmptyPrompt,
  EuiInMemoryTable,
  EuiFlexGroup,
  EuiFlexItem,
  EuiLoadingSpinner,
  EuiModal,
  EuiModalBody,
  EuiModalHeader,
  EuiOverlayMask,
  EuiSpacer,
  EuiTitle,
} from '@elastic/eui';
import { reports_list_users } from './test_data';
import { generateReport } from './main_utils';

const reportStatusOptions = [
  'Created',
  'Error',
  'Pending',
  'Shared',
  'Archived',
];
const reportTypeOptions = ['Schedule', 'Download', 'Alert'];

const emptyMessageReports = (
  <EuiEmptyPrompt
    title={<h3>No reports to display</h3>}
    titleSize="xs"
    body={
      <div>
        <EuiText>
          To get started, share or download a report from a dashboard,
          visualization or saved search, or create a report definition
        </EuiText>
        <EuiText>
          To learn more, see{' '}
          <EuiLink>Get started with Kibana reporting</EuiLink>
        </EuiText>
      </div>
    }
  />
);

export function ReportsTable(props) {
  const { getRowProps, pagination, reportsTableItems, httpClient } = props;

  const [sortField, setSortField] = useState('lastUpdated');
  const [sortDirection, setSortDirection] = useState('des');
  const [showLoading, setShowLoading] = useState(false);

  const handleLoading = (e) => {
    setShowLoading(e);
  };

  const GenerateReportLoadingModal = () => {
    const [isModalVisible, setIsModalVisible] = useState(true);

    const closeModal = () => {
      setIsModalVisible(false);
      setShowLoading(false);
    };
    const showModal = () => setIsModalVisible(true);

    return (
      <div>
        <EuiOverlayMask>
          <EuiModal onClose={closeModal}>
            <EuiModalHeader>
              <EuiTitle>
                <EuiText textAlign="right">
                  <h2>Generating report</h2>
                </EuiText>
              </EuiTitle>
            </EuiModalHeader>
            <EuiModalBody>
              <EuiText>Preparing your file for download...</EuiText>
              <EuiSpacer />
              <EuiFlexGroup justifyContent="spaceAround" alignItems="center">
                <EuiFlexItem>
                  <EuiLoadingSpinner size="xl" />
                </EuiFlexItem>
              </EuiFlexGroup>
            </EuiModalBody>
          </EuiModal>
        </EuiOverlayMask>
      </div>
    );
  };

  const updateMetadata = (url: any) => {
    const onDemandDownloadMetadata = {
      report_name: url['reportName'],
      report_source: url['reportSource'],
      report_type: 'Download',
      description: 'On-demand download of report ' + url['reportName'],
      report_params: {
        url: url['url'],
        window_width: 1440,
        window_height: 2560,
        report_format: 'pdf', // current default format
      },
    };
    return onDemandDownloadMetadata;
  };

  const onDemandDownload = async (url: any) => {
    handleLoading(true);
    await generateReport(updateMetadata(url), httpClient);
    handleLoading(false);
  };

  const reportsTableColumns = [
    {
      field: 'reportName',
      name: 'Name',
      sortable: true,
      truncateText: true,
    },
    {
      field: 'type',
      name: 'Type',
      sortable: true,
      truncateText: false,
    },
    {
      field: 'sender',
      name: 'Sender',
      sortable: true,
      truncateText: false,
    },
    {
      field: 'recipients',
      name: 'Recipient(s)',
      sortable: true,
      truncateText: true,
    },
    {
      field: 'reportSource',
      name: 'Source',
    },
    {
      field: 'lastUpdated',
      name: 'Last updated',
      truncateText: true,
    },
    {
      field: 'state',
      name: 'State',
      sortable: true,
      truncateText: false,
    },
    {
      field: 'url',
      name: 'Download',
      sortable: false,
      actions: [
        {
          name: 'Generate report',
          description: 'Generates the report',
          type: 'icon',
          icon: 'download',
          onClick: (url: any) => onDemandDownload(url),
        },
      ],
    },
  ];

  const sorting = {
    sort: {
      field: sortField,
      direction: sortDirection,
    },
  };

  const uniqueSenders = [];
  reports_list_users.map((user) => {
    if (uniqueSenders.indexOf(user.sender) === -1) {
      uniqueSenders.push(user.sender);
    }
  });

  const uniqueRecipients = [];
  reports_list_users.map((user) => {
    for (const [index, value] of user.recipients.entries()) {
      if (uniqueRecipients.indexOf(value) === -1) {
        uniqueRecipients.push(value);
      }
    }
  });

  const reportsListSearch = {
    box: {
      incremental: true,
    },
    filters: [
      {
        type: 'field_value_selection',
        field: 'sender',
        name: 'Sender',
        multiselect: false,
        options: uniqueSenders.map((user) => ({
          value: user,
          name: user,
          view: user,
        })),
      },
      {
        type: 'field_value_selection',
        field: 'recipients',
        name: 'Recipients',
        multiselect: false,
        options: uniqueRecipients.map((user) => ({
          value: user,
          name: user,
          view: user,
        })),
      },
      {
        type: 'field_value_selection',
        field: 'state',
        name: 'State',
        multiselect: false,
        options: reportStatusOptions.map((state) => ({
          value: state,
          name: state,
          view: state,
        })),
      },
      {
        type: 'field_value_selection',
        field: 'type',
        name: 'Type',
        multiselect: false,
        options: reportTypeOptions.map((type) => ({
          value: type,
          name: type,
          view: type,
        })),
      },
    ],
  };

  const showLoadingModal = showLoading ? <GenerateReportLoadingModal /> : null;

  return (
    <div>
      <EuiInMemoryTable
        items={reportsTableItems}
        itemId="id"
        loading={false}
        message={emptyMessageReports}
        columns={reportsTableColumns}
        search={reportsListSearch}
        pagination={pagination}
        sorting={sorting}
        rowProps={getRowProps}
      />
      {showLoadingModal}
    </div>
  );
}
