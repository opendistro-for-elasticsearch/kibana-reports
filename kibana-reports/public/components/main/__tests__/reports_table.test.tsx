/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { ReportsTable } from '../reports_table';
import httpClientMock from '../../../../test/httpMockClient';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { act } from 'react-dom/test-utils';

const pagination = {
  initialPageSize: 10,
  pageSizeOptions: [8, 10, 13],
};

describe('<ReportsTable /> panel', () => {
  configure({ adapter: new Adapter() });
  test('render component', () => {
    let reportsTableItems = [
      {
        id: '1',
        reportName: 'test report table item',
        type: 'Test type',
        sender: 'N/A',
        recipients: 'N/A',
        reportSource: 'Test report source',
        lastUpdated: 'test updated time',
        state: 'Created',
        url: 'Test url',
      },
    ];
    const { container } = render(
      <ReportsTable
        reportsTableItems={reportsTableItems}
        httpClient={httpClientMock}
        pagination={pagination}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('render empty component', async () => {
    const { container } = render(
      <ReportsTable
        reportsTableItems={[]}
        httpClient={httpClientMock}
        pagination={pagination}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('click on generate button', async () => {
    const promise = Promise.resolve();
    let reportsTableItems = [
      {
        id: '1',
        reportName: 'test report table item',
        type: 'Test type',
        sender: 'N/A',
        recipients: 'N/A',
        reportSource: 'Test report source',
        lastUpdated: 'test updated time',
        state: 'Created',
        url: 'Test url',
      },
    ];

    const component = mount(
      <ReportsTable
        reportsTableItems={reportsTableItems}
        httpClient={httpClientMock}
        pagination={pagination}
      />
    );

    const generateClick = component.find('button').at(6);
    // console.log(generateClick.debug());
    generateClick.simulate('click');
    await act(() => promise);
  });
});
