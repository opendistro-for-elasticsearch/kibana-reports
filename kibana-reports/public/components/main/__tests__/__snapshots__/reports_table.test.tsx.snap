// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportsTable /> panel render component 1`] = `
<div>
  <div
    class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
  >
    <div
      class="euiFlexItem euiSearchBar__searchHolder"
    >
      <div
        class="euiFormControlLayout euiFormControlLayout--fullWidth"
      >
        <div
          class="euiFormControlLayout__childrenWrapper"
        >
          <input
            aria-label="This is a search bar. As you type, the results lower in the page will automatically filter."
            class="euiFieldSearch euiFieldSearch--fullWidth euiFieldSearch-isClearable"
            placeholder="Search..."
            type="search"
            value=""
          />
          <div
            class="euiFormControlLayoutIcons"
          >
            <span
              class="euiFormControlLayoutCustomIcon"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero euiSearchBar__filtersHolder"
    >
      <div
        class="euiFilterGroup"
      >
        <div
          class="euiPopover euiPopover--anchorDownCenter"
          id="field_value_selection_0"
        >
          <div
            class="euiPopover__anchor"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonContent__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
                <span
                  class="euiButtonEmpty__text"
                >
                  <span
                    class="euiFilterButton__textShift"
                    data-text="Type"
                    title="Type"
                  >
                    Type
                  </span>
                </span>
              </span>
            </button>
          </div>
        </div>
        <div
          class="euiPopover euiPopover--anchorDownCenter"
          id="field_value_selection_1"
        >
          <div
            class="euiPopover__anchor"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonContent__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
                <span
                  class="euiButtonEmpty__text"
                >
                  <span
                    class="euiFilterButton__textShift"
                    data-text="State"
                    title="State"
                  >
                    State
                  </span>
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiBasicTable"
  >
    <div>
      <div
        class="euiTableHeaderMobile"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          />
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <div
              class="euiTableSortMobile"
            >
              <div
                class="euiPopover euiPopover--anchorDownRight"
              >
                <div
                  class="euiPopover__anchor"
                >
                  <button
                    class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--flushRight"
                    type="button"
                  >
                    <span
                      class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                    >
                      <svg
                        aria-hidden="true"
                        class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonContent__icon"
                        focusable="false"
                        height="16"
                        role="img"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      />
                      <span
                        class="euiButtonEmpty__text"
                      >
                        Sorting
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <table
        class="euiTable euiTable--responsive euiTable--auto"
        id="random_html_id"
        tabindex="-1"
      >
        <caption
          class="euiScreenReaderOnly euiTableCaption"
        />
        <thead>
          <tr>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_reportName_0"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                  title="Name"
                >
                  Name
                </span>
              </div>
            </th>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_reportSource_1"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                  title="Source"
                >
                  Source
                </span>
              </div>
            </th>
            <th
              aria-live="polite"
              aria-sort="none"
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_type_2"
              role="columnheader"
              scope="col"
            >
              <button
                class="euiTableHeaderButton"
                data-test-subj="tableHeaderSortButton"
                type="button"
              >
                <span
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="Type"
                  >
                    Type
                  </span>
                  <span
                    class="euiScreenReaderOnly"
                  >
                    Click to sort in ascending order
                  </span>
                </span>
              </button>
            </th>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_timeCreated_3"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                  title="Creation time"
                >
                  Creation time
                </span>
              </div>
            </th>
            <th
              aria-live="polite"
              aria-sort="none"
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_state_4"
              role="columnheader"
              scope="col"
            >
              <button
                class="euiTableHeaderButton"
                data-test-subj="tableHeaderSortButton"
                type="button"
              >
                <span
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="State"
                  >
                    State
                  </span>
                  <span
                    class="euiScreenReaderOnly"
                  >
                    Click to sort in ascending order
                  </span>
                </span>
              </button>
            </th>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_id_5"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                  title="Generate"
                >
                  Generate
                </span>
              </div>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="euiTableRow euiTableRow-hasActions"
          >
            <td
              class="euiTableRowCell"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Name
              </div>
              <div
                class="euiTableCellContent euiTableCellContent--overflowingContent"
              >
                <button
                  class="euiLink euiLink--primary"
                  id="reportDetailsLink"
                  type="button"
                >
                  test report table item
                </button>
              </div>
            </td>
            <td
              class="euiTableRowCell"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Source
              </div>
              <div
                class="euiTableCellContent euiTableCellContent--overflowingContent"
              >
                <a
                  class="euiLink euiLink--primary"
                  href="Test url"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Test report source
                </a>
              </div>
            </td>
            <td
              class="euiTableRowCell"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Type
              </div>
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                >
                  Test type
                </span>
              </div>
            </td>
            <td
              class="euiTableRowCell"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Creation time
              </div>
              <div
                class="euiTableCellContent euiTableCellContent--overflowingContent"
              >
                <div
                  class="euiText euiText--small"
                >
                  Invalid Date @ Invalid Date
                </div>
              </div>
            </td>
            <td
              class="euiTableRowCell"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                State
              </div>
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                >
                  Created
                </span>
              </div>
            </td>
            <td
              class="euiTableRowCell"
            >
              <div
                class="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
              >
                Generate
              </div>
              <div
                class="euiTableCellContent euiTableCellContent--overflowingContent"
              >
                <button
                  class="euiLink euiLink--primary"
                  type="button"
                >
                   
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon-isLoading"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div>
      <div
        class="euiSpacer euiSpacer--m"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiPopover euiPopover--anchorUpRight euiPopover--withTitle"
          >
            <div
              class="euiPopover__anchor"
            >
              <button
                class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall"
                data-test-subj="tablePaginationPopoverButton"
                type="button"
              >
                <span
                  class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonContent__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                  <span
                    class="euiButtonEmpty__text"
                  >
                    Rows per page
                    : 
                    10
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <nav
            class="euiPagination"
          >
            <button
              aria-label="Previous page"
              class="euiButtonIcon euiButtonIcon--text"
              data-test-subj="pagination-button-previous"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              />
            </button>
            <ul
              class="euiPagination__list"
            >
              <li
                class="euiPagination__item"
              >
                <button
                  aria-controls="random_html_id"
                  aria-current="true"
                  aria-label="Page 1 of 1"
                  class="euiButtonEmpty euiButtonEmpty--text euiButtonEmpty--xSmall euiButtonEmpty-isDisabled euiPaginationButton euiPaginationButton-isActive euiPaginationButton--hideOnMobile"
                  data-test-subj="pagination-button-0"
                  disabled=""
                  type="button"
                >
                  <span
                    class="euiButtonContent euiButtonEmpty__content"
                  >
                    <span
                      class="euiButtonEmpty__text"
                    >
                      1
                    </span>
                  </span>
                </button>
              </li>
            </ul>
            <button
              aria-label="Next page"
              class="euiButtonIcon euiButtonIcon--text"
              data-test-subj="pagination-button-next"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              />
            </button>
          </nav>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ReportsTable /> panel render empty component 1`] = `
<div>
  <div
    class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive euiFlexGroup--wrap"
  >
    <div
      class="euiFlexItem euiSearchBar__searchHolder"
    >
      <div
        class="euiFormControlLayout euiFormControlLayout--fullWidth"
      >
        <div
          class="euiFormControlLayout__childrenWrapper"
        >
          <input
            aria-label="This is a search bar. As you type, the results lower in the page will automatically filter."
            class="euiFieldSearch euiFieldSearch--fullWidth euiFieldSearch-isClearable"
            placeholder="Search..."
            type="search"
            value=""
          />
          <div
            class="euiFormControlLayoutIcons"
          >
            <span
              class="euiFormControlLayoutCustomIcon"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.271 11.978l3.872 3.873a.502.502 0 00.708 0 .502.502 0 000-.708l-3.565-3.564c2.38-2.747 2.267-6.923-.342-9.532-2.73-2.73-7.17-2.73-9.898 0-2.728 2.729-2.728 7.17 0 9.9a6.955 6.955 0 004.949 2.05.5.5 0 000-1 5.96 5.96 0 01-4.242-1.757 6.01 6.01 0 010-8.486c2.337-2.34 6.143-2.34 8.484 0a6.01 6.01 0 010 8.486.5.5 0 00.034.738z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero euiSearchBar__filtersHolder"
    >
      <div
        class="euiFilterGroup"
      >
        <div
          class="euiPopover euiPopover--anchorDownCenter"
          id="field_value_selection_0"
        >
          <div
            class="euiPopover__anchor"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiButtonContent__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.069 5.157L8.384 9.768a.546.546 0 01-.768 0L2.93 5.158a.552.552 0 00-.771 0 .53.53 0 000 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 000-.76.552.552 0 00-.771 0z"
                    fill-rule="non-zero"
                  />
                </svg>
                <span
                  class="euiButtonEmpty__text"
                >
                  <span
                    class="euiFilterButton__textShift"
                    data-text="Type"
                    title="Type"
                  >
                    Type
                  </span>
                </span>
              </span>
            </button>
          </div>
        </div>
        <div
          class="euiPopover euiPopover--anchorDownCenter"
          id="field_value_selection_1"
        >
          <div
            class="euiPopover__anchor"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--text euiFilterButton euiFilterButton--hasIcon"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiButtonContent__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.069 5.157L8.384 9.768a.546.546 0 01-.768 0L2.93 5.158a.552.552 0 00-.771 0 .53.53 0 000 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 000-.76.552.552 0 00-.771 0z"
                    fill-rule="non-zero"
                  />
                </svg>
                <span
                  class="euiButtonEmpty__text"
                >
                  <span
                    class="euiFilterButton__textShift"
                    data-text="State"
                    title="State"
                  >
                    State
                  </span>
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiBasicTable"
  >
    <div>
      <div
        class="euiTableHeaderMobile"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          />
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <div
              class="euiTableSortMobile"
            >
              <div
                class="euiPopover euiPopover--anchorDownRight"
              >
                <div
                  class="euiPopover__anchor"
                >
                  <button
                    class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--flushRight"
                    type="button"
                  >
                    <span
                      class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                    >
                      <svg
                        aria-hidden="true"
                        class="euiIcon euiIcon--medium euiButtonContent__icon"
                        focusable="false"
                        height="16"
                        role="img"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13.069 5.157L8.384 9.768a.546.546 0 01-.768 0L2.93 5.158a.552.552 0 00-.771 0 .53.53 0 000 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 000-.76.552.552 0 00-.771 0z"
                          fill-rule="non-zero"
                        />
                      </svg>
                      <span
                        class="euiButtonEmpty__text"
                      >
                        Sorting
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <table
        class="euiTable euiTable--responsive euiTable--auto"
        id="random_html_id"
        tabindex="-1"
      >
        <caption
          class="euiScreenReaderOnly euiTableCaption"
        />
        <thead>
          <tr>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_reportName_0"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                  title="Name"
                >
                  Name
                </span>
              </div>
            </th>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_reportSource_1"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                  title="Source"
                >
                  Source
                </span>
              </div>
            </th>
            <th
              aria-live="polite"
              aria-sort="none"
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_type_2"
              role="columnheader"
              scope="col"
            >
              <button
                class="euiTableHeaderButton"
                data-test-subj="tableHeaderSortButton"
                type="button"
              >
                <span
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="Type"
                  >
                    Type
                  </span>
                  <span
                    class="euiScreenReaderOnly"
                  >
                    Click to sort in ascending order
                  </span>
                </span>
              </button>
            </th>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_timeCreated_3"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                  title="Creation time"
                >
                  Creation time
                </span>
              </div>
            </th>
            <th
              aria-live="polite"
              aria-sort="none"
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_state_4"
              role="columnheader"
              scope="col"
            >
              <button
                class="euiTableHeaderButton"
                data-test-subj="tableHeaderSortButton"
                type="button"
              >
                <span
                  class="euiTableCellContent"
                >
                  <span
                    class="euiTableCellContent__text"
                    title="State"
                  >
                    State
                  </span>
                  <span
                    class="euiScreenReaderOnly"
                  >
                    Click to sort in ascending order
                  </span>
                </span>
              </button>
            </th>
            <th
              class="euiTableHeaderCell"
              data-test-subj="tableHeaderCell_id_5"
              role="columnheader"
              scope="col"
            >
              <div
                class="euiTableCellContent"
              >
                <span
                  class="euiTableCellContent__text"
                  title="Generate"
                >
                  Generate
                </span>
              </div>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="euiTableRow"
          >
            <td
              class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
              colspan="6"
            >
              <div
                class="euiTableCellContent euiTableCellContent--alignCenter"
              >
                <span
                  class="euiTableCellContent__text"
                >
                  <div
                    class="euiEmptyPrompt"
                  >
                    <span
                      class="euiTextColor euiTextColor--subdued"
                    >
                      <h3
                        class="euiTitle euiTitle--xsmall"
                      >
                        No reports to display
                      </h3>
                      <div
                        class="euiSpacer euiSpacer--m"
                      />
                      <div
                        class="euiText euiText--medium"
                      >
                        <div>
                          <div
                            class="euiText euiText--medium"
                          >
                            Create a report definition, or share/download a report from a dashboard, saved search or visualization.
                          </div>
                          <div
                            class="euiText euiText--medium"
                          >
                            To learn more, see
                             
                            <button
                              class="euiLink euiLink--primary"
                              type="button"
                            >
                              Get started with Kibana reporting 
                              <svg
                                aria-hidden="true"
                                class="euiIcon euiIcon--medium euiIcon-isLoading"
                                focusable="false"
                                height="16"
                                role="img"
                                viewBox="0 0 16 16"
                                width="16"
                                xmlns="http://www.w3.org/2000/svg"
                              />
                            </button>
                          </div>
                        </div>
                      </div>
                    </span>
                  </div>
                </span>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
